### PR TITLE
Upgrade no-newlines to no-collapsible-whitespace

### PR DESCRIPTION
**Pull request by deBhal**
This PR upgrades the `i18n-no-newlines` rule to a more general `i18n-no-collapsible-whitespace` to check for whitespace that HTML rendering would strip, namely tabs, newlines and consecutive spaces.

The motivation for this new rule is the same as the `no-newlines` rule, and it includes all the previous issues, so I think it makes sense to replace the old rule, rather than add a new overlapping rule.

However, dropping `i18n-no-newlines` is a breaking change, so I'd suggest that we include both rules until the next major release.

To that end, I've made an "upgrade" commit that adds the new rule and deletes the old one, and a second commit that restores the `i18n-no-newlines` rule that we should be able to revert when we do the next major release.

### CODE REVIEW

File: tests/util/format-message.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * Given a message containing data terms, format the string using the specified
+ * terms object.
+ *
+ * @see https://github.com/eslint/eslint/blob/v2.12.0/lib/eslint.js#L964-L971
+ *
+ * @param  {String} message Message template
+ * @param  {Object} terms   Terms
+ * @return {String}         Formatted message
+ */
+function formatMessage( message, terms ) {
```
Comment (aduth) [inline tests/util/format-message.js:18]: Should we add tests for this?

**Comment by aduth** [comment]
Another breaking change I'd intended to have included in 2.0 was the removal of the `no-lodash-import` rule. I'd not be opposed to pushing a 3.0 release in the near future (per semver, we shouldn't be "afraid" of major version bumps) including the removal of both these rules.

File: tests/util/format-message.js (hunk 1 above)
Comment (deBhal) [reply tests/util/format-message.js:18]: It gets exercised by every test that uses it and I think we could get away without tests, so I've added a couple of simple ones.  ([yo dawg](https://cdn.meme.am/instances/500x/59236929.jpg)]

**Comment by deBhal** [comment]
We should capture these pending changes somewhere.  It would fit in an issue or a milestone, but perhaps we should open a PR and add them to that.  Leaving PRs open for too long is distasteful, so it might line up pretty well with not being afraid of bumping the version.

What do you think?

File: tests/util/format-message.js (hunk 1 above)
Comment (aduth) [reply tests/util/format-message.js:18]: > (yo dawg]

😆  You might appreciate the latest changelog notes: https://github.com/Automattic/eslint-plugin-wpcalypso/blob/master/CHANGELOG.md#v200-august-24-2016

File: tests/lib/rules/i18n-no-collapsible-whitespace.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+var rule = require( '../../../lib/rules/i18n-no-collapsible-whitespace' ),
+	config = { env: { es6: true } },  // support for string templates
+	formatMessage = require( '../../util/format-message' ),
+	RuleTester = require( 'eslint' ).RuleTester;
+
+//------------------------------------------------------------------------------
+// Constants
+//------------------------------------------------------------------------------
+
+const NO_NEWLINES = formatMessage( rule.ERROR_MESSAGE, { problem: ' (\\n)' } );
```
Comment (aduth) [inline tests/lib/rules/i18n-no-collapsible-whitespace.js]: I guess now that we're running Node 4+ in the project, `const` should be okay. That said, I'd still prefer consistently at least within the same file, where there are `var` declarations above that could be `const`.

**Comment by aduth** [comment]
I create a v3.0.0 milestone and a separate pull request to remove the Lodash import rule (#26). I'd say we should just go ahead and remove/replace the existing `i18n-no-newlines` rule as part of this pull request.

Another possible change to include in the next release could be enhancements to our `jsx-classname-namespace` rule pending [discussion on the expected format of non-root class names](https://github.com/Automattic/wp-calypso/pull/7866/files/1cb3beb74615810026453bb6204a051489fad604#r77802780).

File: tests/lib/rules/i18n-no-collapsible-whitespace.js (hunk 2 above)
Comment (deBhal) [reply tests/lib/rules/i18n-no-collapsible-whitespace.js]: Let's stick with var, at least for now.  We essentially document that that's what we're doing in `.eslintrc.json`.

**Comment by deBhal** [comment]
I just want to note that in preferring `var` I meant "in this PR" not "at this point in time".  I'd be in favour or switching over to const and let, but it doesn't seem like a high priority

**Pull request merged**